code/modules/mob/living/simple_animal/friendly/diona.dm

```diff
@@ -102,61 +104,25 @@
 	set name = "Split from gestalt"
 	set desc = "Split away from your gestalt as a lone nymph."
 
-	if(!(istype(src.loc,/mob/living/carbon)))
-		src.verbs -= /mob/living/simple_animal/diona/proc/split
+	if(!(iscarbon(loc)))
+		verbs -= /mob/living/simple_animal/diona/proc/split
 		return
 
-	to_chat(src.loc, "You feel a pang of loss as [src] splits away from your biomass.")
-	to_chat(src, "You wiggle out of the depths of [src.loc]'s biomass and plop to the ground.")
+	to_chat(loc, "You feel a pang of loss as [src] splits away from your biomass.")
+	to_chat(src, "You wiggle out of the depths of [loc]'s biomass and plop to the ground.")
 
-	var/mob/living/M = src.loc
+	var/mob/living/M = loc
 
-	src.loc = get_turf(src)
-	src.verbs -= /mob/living/simple_animal/diona/proc/split
-	src.verbs += /mob/living/simple_animal/diona/proc/merge
+	forceMove(get_turf(src))
+	verbs -= /mob/living/simple_animal/diona/proc/split
+	verbs += /mob/living/simple_animal/diona/proc/merge
 
 	if(istype(M))
 		for(var/atom/A in M.contents)
-			if(istype(A,/mob/living/simple_animal/borer) || istype(A,/obj/item/weapon/holder))
+			if(istype(A, /mob/living/simple_animal/borer) || istype(A, /obj/item/weapon/holder))
 				return
 	M.status_flags &= ~PASSEMOTES
 
-/mob/living/simple_animal/diona/verb/fertilize_plant()
-	set category = "Diona"
-	set name = "Fertilize plant"
-	set desc = "Turn your food into nutrients for plants."
-
-	var/list/trays = list()
-	for(var/obj/machinery/hydroponics/tray in range(1))
-		if(tray.nutrilevel < 10)
-			trays += tray
-
-	var/obj/machinery/hydroponics/target = input("Select a tray:") as null|anything in trays
-
-	if(!src || !target || target.nutrilevel == 10) return //Sanity check.
-
-	src.nutrition -= ((10-target.nutrilevel)*5)
-	target.adjustNutri(10)
-	src.visible_message("<span class='danger'>[src] secretes a trickle of green liquid from its tail, refilling [target]'s nutrient tray.","\red You secrete a trickle of green liquid from your tail, refilling [target]'s nutrient tray.</span>")
-
-/mob/living/simple_animal/diona/verb/eat_weeds()
-	set category = "Diona"
-	set name = "Eat Weeds"
-	set desc = "Clean the weeds out of soil or a hydroponics tray."
-
-	var/list/trays = list()
-	for(var/obj/machinery/hydroponics/tray in range(1))
-		if(tray.weedlevel > 0)
-			trays += tray
-
-	var/obj/machinery/hydroponics/target = input("Select a tray:") as null|anything in trays
-
-	if(!src || !target || target.weedlevel == 0) return //Sanity check.
-
-	src.nutrition += target.weedlevel * 15
-	target.adjustWeeds(-10)
-	src.visible_message("<span class='danger'>[src] begins rooting through [target], ripping out weeds and eating them noisily.</span>","<span class='danger'>You begin rooting through [target], ripping out weeds and eating them noisily.</span>")
-
```

Comment:
The removal of the above two verbs are the only mechanics changes in this file (and that functionality has been moved to `/obj/machinery/hydroponics/attack_animal(mob/user)`